### PR TITLE
Added hashes to check all files are copied

### LINKED ISSUE #5: Finish event not fired ?
Hi,

I'm using Vinyl File to represents a file stream in my application. I use clones to pipe the same content to multiple ffmpeg commands (using stream-transcoder or fluent-ffmpeg (same behaviour happening)).

It works great for the first original Vinyl file, but not for the clones. 
I've isolated the error in the example below, if I run it multiple times, sometimes I got the 3 finish events, sometimes I just have the one for the original stream. But the files are really copied in the example.

```
const cloneable = require('cloneable-readable');
const path = require('path');
const fs = require('fs');

let stream = cloneable(fs.createReadStream(path.join(__dirname, 'sample.wav')));

function pipe(stream, num) {

    stream.on('end', () => {
        console.log(`Reaching end of stream : ${num}`);
    });

    stream.pipe(fs.createWriteStream(path.join(__dirname, `sample-${num}.wav`))).on('finish', () => {
        console.log(`Stop writing to file for stream : ${num}`);
    });

}

setImmediate(pipe.bind(null, stream, 1)); // Pipe in another event loop tick <-- this one finished only, it's the original cloneable.
pipe(stream.clone(), 0);    // Pipe in the same event loop tick
setTimeout(pipe.bind(null, stream.clone(), 2), 1000);   // Pipe a long time after

setTimeout(() => { }, 2000);    // here to maintain a ref in the event loop longer.
```

Here is an example of what I can see during my tests : 

screen shot 2018-03-03 at 18 48 21

### Environment

I'm using Node v6.10.3 on mac.
I use a sample wav file of 450kb.

## Full example

I think it's not an ffmpeg issue, but for your interest, here is an example of code using ffmpeg which is not working : 

```
const path = require('path');
const fs = require('fs');
const cloneable = require('cloneable-readable');
const ffmpeg = require('fluent-ffmpeg');

let stream = cloneable(fs.createReadStream(path.join(__dirname, 'sample.wav')));

function pipe(stream, num) {

    stream.on('end', () => {
        console.log(`End writing the stream : ${num}`);
    });

    ffmpeg(stream)
        .audioFrequency(8000)
        .format('s16le')
        .output(path.join(__dirname, 'copy', `sample-${num}.wav`))
        .on('end', () => {
            console.log(`finished for the stream: ${num}`);
        })
        .run();

}

setImmediate(pipe.bind(null, stream, 1)); // Pipe in another event loop tick <-- it's the original cloneable.
pipe(stream.clone(), 0);    // Pipe in the same event loop tick
setTimeout(pipe.bind(null, stream.clone(), 2), 1000);   // Pipe a long time after

setTimeout(() => { }, 2000);    // here to maintain a ref in the event loop longer.
```

If you run the process above, it will never exit. Events are still attached in the event loop ? Are they lost somewhere ? 

Thank you for the time and the response, 
Pierre.

### PR DESCRIPTION
**Pull request by mcollina**
Fixes #5.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -50,18 +54,34 @@ Cloneable.prototype.clone = function () {
...
+  }
 
   return clone
 }
 
+Cloneable.prototype._destroy = function (err, cb) {
+  if (!err) {
+    this.push(null)
+    this.end()
+    this.emit('close')
```
Comment (mafintosh) [inline index.js]: note this will break when node master is released

Comment (mafintosh) [reply index.js]: breakage meaning close will be emitted twice

File: index.js [hunk 2]
```diff
@@ -119,4 +139,15 @@ Cloneable.isCloneable = function (stream) {
   return stream instanceof Cloneable || stream instanceof Clone
 }
 
+Clone.prototype._destroy = function (err, cb) {
+  if (!err) {
+    this.push(null)
+    this.end()
+    this.emit('close')
```
Comment (mafintosh) [inline index.js]: same as above

File: index.js [hunk 3]
```diff
@@ -119,4 +139,15 @@ Cloneable.isCloneable = function (stream) {
...
+  }
+
+  console.log('clone destroy', err)
```
Comment (mafintosh) [inline index.js]: did you mean to add this log?

File: index.js [hunk 4]
```diff
@@ -119,4 +139,15 @@ Cloneable.isCloneable = function (stream) {
   return stream instanceof Cloneable || stream instanceof Clone
 }
 
+Clone.prototype._destroy = function (err, cb) {
+  if (!err) {
+    this.push(null)
+    this.end()
```
Comment (mafintosh) [inline index.js]: i'd get rid of the push(null) and end() calls. this makes error handling *super* tricky as the stream will look like it ended gracefully (unless that's what you are going for of course)

File: index.js [hunk 5]
```diff
@@ -50,18 +54,34 @@ Cloneable.prototype.clone = function () {
...
+    this.on('newListener', onData)
+  }
 
   return clone
 }
 
+Cloneable.prototype._destroy = function (err, cb) {
+  if (!err) {
+    this.push(null)
+    this.end()
```
Comment (mafintosh) [inline index.js]: i'd get rid of the push(null) and end() calls. this makes error handling *super* tricky as the stream will look like it ended gracefully (unless that's what you are going for of course)

File: index.js (hunk 1 above)
Comment (mcollina) [reply index.js]: We are relying on readable-stream@2 here, and that would end up in readable-stream@3. So it's safe to do it this way atm.

File: index.js (hunk 4 above)
Comment (mcollina) [reply index.js]: Are you sure? We are doing these only if there was no error.

File: index.js (hunk 1 above)
Comment (mafintosh) [reply index.js]: :+1:

File: index.js (hunk 4 above)
Comment (mafintosh) [reply index.js]: Depends on your usecase. i'd *never* do it. Plenty of things do .destroy() to just teardown the pipeline without emitting errors

Comment (mcollina) [reply index.js]: This module is relying heavily on that behavior to work properly. Unfortunately it's a massive rewrite and this fix an important bug.

**Pull request merged**